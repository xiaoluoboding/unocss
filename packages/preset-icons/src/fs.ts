import { promises as fs } from 'fs'
import { isPackageExists, resolveModule } from 'local-pkg'
import type { IconifyJSON } from '@iconify/types'

const _collections: Record<string, Promise<IconifyJSON | undefined>> = {}

const isLegacyExists = isPackageExists('@iconify/json')

export async function loadCollectionFromFS(name: string) {
  if (!_collections[name])
    _collections[name] = task()
  return _collections[name]

  async function task() {
    let jsonPath = resolveModule(`@iconify-json/${name}/icons.json`)
    if (!jsonPath && isLegacyExists)
      jsonPath = resolveModule(`@iconify/json/json/${name}.json`)

    if (jsonPath) {
      const icons = JSON.parse(await fs.readFile(jsonPath, 'utf8'))
      return icons
    }
    else {
      return undefined
    }
  }
}
