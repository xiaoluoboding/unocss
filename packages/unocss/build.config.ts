import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  entries: [
    'src/index',
    'src/vite',
    'src/preset-uno',
    'src/preset-icons',
    'src/preset-attributify',
  ],
  clean: true,
  declaration: true,
  externals: [
    'vite',
  ],
})
