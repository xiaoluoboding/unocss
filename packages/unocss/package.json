{
  "name": "unocss",
  "version": "0.16.4",
  "description": "The instant on-demand Atomic CSS engine.",
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "homepage": "https://github.com/antfu/unocss#readme",
  "bugs": {
    "url": "https://github.com/antfu/unocss/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/antfu/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "license": "MIT",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.ts"
    },
    "./vite": {
      "require": "./dist/vite.cjs",
      "import": "./dist/vite.mjs",
      "types": "./dist/vite.d.ts"
    }
  },
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "files": [
    "dist",
    "*.d.ts"
  ],
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "dependencies": {
    "@unocss/core": "workspace:*",
    "@unocss/cli": "workspace:*",
    "@unocss/reset": "workspace:*",
    "@unocss/preset-icons": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "engines": {
    "node": ">=14"
  }
}
