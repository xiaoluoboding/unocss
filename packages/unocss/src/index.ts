import type { UserConfig } from '@unocss/core'

export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
export { default as presetIcons } from '@unocss/preset-icons'

export function defineConfig<Theme extends {}>(config: UserConfig<Theme>) {
  return config
}
