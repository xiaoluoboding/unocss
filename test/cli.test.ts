import { resolve } from 'path'
import fs from 'fs-extra'
import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import execa from 'execa'

export const cacheDir = resolve('.cache')
export const cli = resolve(__dirname, '../packages/cli/src/cli.ts')

beforeAll(async() => {
  await fs.remove(cacheDir)
})

afterAll(async() => {
  await fs.remove(cacheDir)
})

describe('cli', () => {
  it('builds uno.css', async() => {
    const { output } = await runCli({
      'views/index.html': '<div class="p-4 max-w-screen-md"></div>',
    })

    expect(output).toMatchSnapshot()
  })

  it('supports unocss.config.js', async() => {
    const { output } = await runCli({
      'views/index.html': '<div class="box"></div>',
      'unocss.config.js': `
      import { defineConfig } from 'unocss'
      export default defineConfig({
        shortcuts: [{ box: 'max-w-7xl mx-auto bg-gray-100 rounded-md shadow-sm p-4' }]
      })
    `,
    })

    expect(output).toMatchSnapshot()
  })
})

async function runCli(files: Record<string, string>) {
  const testDir = resolve(cacheDir, Date.now().toString())

  await Promise.all(
    Object.entries(files).map(([path, content]) =>
      fs.outputFile(resolve(testDir, path), content, 'utf8'),
    ),
  )

  const { exitCode, stdout, stderr } = await execa('npx', ['esno', cli, 'views/**/*'], {
    cwd: testDir,
    // stdio: 'inherit',
  })

  const logs = stdout + stderr
  if (exitCode !== 0)
    throw new Error(logs)

  const output = await fs.readFile(resolve(testDir, 'uno.css'), 'utf8')

  return {
    output,
    logs,
  }
}
