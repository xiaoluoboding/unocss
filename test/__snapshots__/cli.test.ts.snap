// Vitest Snapshot v1

exports[`cli > builds uno.css 1`] = `
"/* layer: default */
.max-w-screen-md{max-width:768px;}
.p-4{padding:1rem;}"
`;

exports[`cli > supports unocss.config.js 1`] = `
"/* layer: shortcuts */
.box{margin-left:auto;margin-right:auto;max-width:80rem;border-radius:0.375rem;--un-bg-opacity:1;background-color:rgba(243,244,246,var(--un-bg-opacity));padding:1rem;--un-shadow-color:0,0,0;--un-shadow:0 1px 2px 0 rgba(var(--un-shadow-color), 0.05);box-shadow:var(--un-ring-offset-shadow, 0 0 #0000), var(--un-ring-shadow, 0 0 #0000), var(--un-shadow);}"
`;
